src/CSESoftware.Core/Entity/BaseEntity.cs

```diff
@@ -1,22 +1,25 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Globalization;
 
 namespace CSESoftware.Core.Entity
 {
-    public abstract class BaseEntity<T> : IBaseEntity<T>
+    public abstract class BaseEntity<T> : EntityWithId<T>, IBaseEntity<T>
```

Comment:
Maybe I'm wrong, but it seems like only IModifiedEntity declares CreatedDate and ModifiedDate but we're using the properties here and we're not implementing this interface, am I missing something?

Reply:
And if that is correct, should we be moving the CreateSetup and UpdateSetup methods onto at least the IActiveEntity? seems like the lower interfaces are exposing it but the concrete classes aren't using it

Reply:
IBaseEntity has IModifiedEntity

Reply:
Ah I missed that, good catch